nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/migration/ControllerServiceFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.migration;
+
+import org.apache.nifi.controller.service.ControllerServiceNode;
+
+import java.util.Map;
+
+public interface ControllerServiceFactory {
+    ControllerServiceCreationDetails creationDetails(String implementationClassName, Map<String, String> propertyValues);
```

Comment:
Minor naming question, is there a reason for naming this `creationDetails` as opposed to `getCreationDetails()`?

Reply:
Nope - `getCreationDetails` is probably a better name. Will change.